### PR TITLE
Api/Bucket-Policy: Handler tests

### LINKED ISSUE #1990: tests: Address poor coverage for bucket policy tests.
bucket-handlers.go: 29% 
bucket-policy-handlers.go: 39%

### PR DESCRIPTION
**Pull request by hackintoshrao**
fixes https://github.com/minio/minio/issues/1990

### CODE REVIEW

File: test-utils_test.go [hunk 1]
```diff
@@ -197,6 +197,8 @@ func newTestRequest(method, urlStr string, contentLength int64, body io.ReadSeek
 		body.Seek(0, 0)
 		// Add body
 		req.Body = ioutil.NopCloser(body)
+	} else {
+		req.Body = ioutil.NopCloser(bytes.NewReader([]byte("")))
```
Comment (harshavardhana) [inline test-utils_test.go:204]: why is this necessary?

Comment (hackintoshrao) [inline test-utils_test.go:204]: @harshavardhana : There was a crash while reading the req.Body in `ioutil.ReadAll` inside `authHandler`. This fixed it.

Comment (harshavardhana) [inline test-utils_test.go:204]: Can paste the crash this might be masking it..

**Comment by hackintoshrao** [comment]
Produced by removing the `else` condition and running `go test -run=TestDeleteBucketPolicyHandler .`

```

--- FAIL: TestDeleteBucketPolicyHandler (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x2543c9]

goroutine 7 [running]:
panic(0x763980, 0xc82000e0a0)
    /usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.tRunner.func1(0xc82007eb40)
    /usr/local/go/src/testing/testing.go:467 +0x192
panic(0x763980, 0xc82000e0a0)
    /usr/local/go/src/runtime/panic.go:426 +0x4e9
io/ioutil.readAll.func1(0xc820173688)
    /usr/local/go/src/io/ioutil/ioutil.go:30 +0x11e
panic(0x763980, 0xc82000e0a0)
    /usr/local/go/src/runtime/panic.go:426 +0x4e9
bytes.(*Buffer).ReadFrom(0xc8201735e0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/bytes/buffer.go:176 +0x239
io/ioutil.readAll(0x0, 0x0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/io/ioutil/ioutil.go:33 +0x156
io/ioutil.ReadAll(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/io/ioutil/ioutil.go:42 +0x51
github.com/minio/minio.isReqAuthenticated(0xc820162700, 0x4)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/auth-handler.go:125 +0x90
github.com/minio/minio.objectAPIHandlers.DeleteBucketPolicyHandler(0xf45020, 0xc820161a10, 0xf45458, 0xc82024ca00, 0xc820162700)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/bucket-policy-handlers.go:245 +0xb8
github.com/minio/minio.(objectAPIHandlers).DeleteBucketPolicyHandler-fm(0xf45458, 0xc82024ca00, 0xc820162700)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/api-router.go:78 +0x4a
net/http.HandlerFunc.ServeHTTP(0xc820185300, 0xf45458, 0xc82024ca00, 0xc820162700)
    /usr/local/go/src/net/http/server.go:1618 +0x3a
github.com/minio/minio/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc820013950, 0xf45458, 0xc82024ca00, 0xc820162700)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/vendor/github.com/gorilla/mux/mux.go:100 +0x29e
github.com/minio/minio.testDeleteBucketPolicyHandler(0xf45020, 0xc820161a10, 0x845740, 0xa, 0xc82007eb40)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/bucket-policy-handlers_test.go:664 +0x137c
github.com/minio/minio.ExecObjectLayerTest(0xc82007eb40, 0x993d50)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/test-utils_test.go:615 +0x1a3
github.com/minio/minio.TestDeleteBucketPolicyHandler(0xc82007eb40)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/bucket-policy-handlers_test.go:548 +0x2d
testing.tRunner(0xc82007eb40, 0xca4870)
    /usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
    /usr/local/go/src/testing/testing.go:582 +0x892
```

**Comment by harshavardhana** [comment]
LGTM, Thanks :+1:
